plugins/trace_api_plugin/include/eosio/trace_api_plugin/data_log.hpp

```diff
@@ -10,4 +10,34 @@ namespace eosio { namespace trace_api_plugin {
       block_trace_v0
    >;
 
+   /**
+    * append an entry to the end of the slice
+    *
+    * @param entry : the entry to append
+    * @param slice : the slice to append entry to
+    * @return the offset in the slice where that entry is written
+    */
+   template<typename DataEntry, typename Slice>
+   uint64_t append_data_log(const DataEntry &entry, Slice &slice) {
+      auto data = fc::raw::pack(entry);
+      const auto offset = slice.tellp();
+      slice.write(data.data(), data.size());
+      slice.flush();
+      slice.sync();
+      return offset;
+   }
+
+   /**
+    * extract an entry from the data log
+    *
+    * @param slice : the slice to extract entry from
+    * @return the extracted data log
+    */
+   template<typename DataEntry, typename Slice>
+   DataEntry extract_data_log( Slice& slice ) {
+      DataEntry entry;
+      auto ds = slice.create_datastream();
+      fc::raw::unpack(ds, entry);
+      return entry;
+   }
```

Comment:
I think it would be less confusing to name these template params `File` instead `Slice`. Also the template method `extract_metadata_log`.

Reply:
it can extract anything since it is templated, so metadata would be too restrictive

Reply:
I'm only saying rename `Slice` => `File` for `extract_data_log` & `extract_metadata_log`